lib/gears/filesystem.lua

```diff
@@ -11,20 +11,30 @@ local gtable = require("gears.table")
 
 local filesystem = {}
 
+local function make_parent(gfile)
+    local success, err = gfile:make_directory_with_parents()
+    if success then
+        return true
+    end
+    if err.domain == Gio.IOErrorEnum and err.code == "EXISTS" then
+        -- Directory already exists, let this count as success
+        return true
+    end
+    return false, err
+end
+
 --- Create a directory
 -- @tparam string dir The directory.
 -- @return (true, nil) on success, (false, err) on failure
 function filesystem.mkdir(dir)
```

Comment:
That's true. I do see that `Gio.File` has `make_directory` and `make_directory_with_parents`.

I think that would be a good change since we're not calling `os.mkdir` anymore. Also, to fit with the other `get_*_directory` functions perhaps expand `make_parent_dir` to `make_parent_directories`?